### PR TITLE
Add support for regnet and stock and flow

### PR DESCRIPTION
**Pull request by brandomr**
This PR updates the `mira_model` context and the `mira_config_edit` contexts to support loading, editing and saving regnets and stock and flows. In both contexts the strategy is to store the `schema_name` on the `context` (`self.schema_name`) then pass this information throughout the context, agent and procedures to ensure that the frameworks are handled appropriately.

> Notably, the stratification prompt had to be updated to ensure that if the stratification is to be performed on a `regnet` the `structure` argument is specified to be an empty array `[]` per [Ben's guidance](https://github.com/gyorilab/mira/issues/275#issuecomment-1915952951). See screenshot below for confirmation that this is now handled properly

Screenshot 2024-01-30 at 10 08 08 AM

This PR also includes updates to interact with HMI Server (not TDS) for the contexts referenced as well as for datasets and decapodes (untested).

### CODE REVIEW

File: src/askem_beaker/contexts/dataset/context.py [hunk 1]
```diff
@@ -40,9 +40,12 @@ async def set_datasets(self, dataset_map, parent_header={}):
             elif isinstance(dataset_map_item, dict):
                 dataset_id = dataset_map_item["id"]
             else:
-                raise ValueError("Unable to parse dataset mapping")
-            meta_url = f"{os.environ['DATA_SERVICE_URL']}/datasets/{dataset_id}"
-            dataset_info_req = requests.get(meta_url)
+                raise ValueError("Unable to parse dataset mapping")      
+            meta_url = f"{os.environ['HMI_SERVER_URL']}/datasets/{dataset_id}"
```
Comment (Tom-Szendrey) [inline src/askem_beaker/contexts/dataset/context.py:44]: Do we not need to change the docker-compose.yaml environment.DATA_SERVICE_URL -> HMI_SERVER_URL for this?
I may be misunderstanding

Comment (brandomr) [reply src/askem_beaker/contexts/dataset/context.py:44]: Ah so I updated the `env.example` and for development ive just used staging/beta to confirm that thing were working as expected but for true local end to end development we need to update the `docker-compose.yaml`.

When I'm running it I just use a `docker-compose-dev.yaml` like:

```
---
# Jupyter LLM docker compose by Matthew Printz

version: '3'
services:
  jupyter:
    build:
      dockerfile: Dockerfile
      context: .
    ports:
      - "8888:8888"
    environment:
      - DEBUG=1
      - JUPYTER_SERVER=http://jupyter:8888
      - JUPYTER_TOKEN=89f73481102c46c0bc13b2998f9a4fce
      - ENABLE_USER_PROMPT=true
      - OPENAI_API_KEY
    env_file:
      - .env      
    volumes:
      - ./src/askem_beaker:/home/jupyter/askem_beaker
      - .:/jupyter
    command: ["python", "-m", "beaker_kernel.server.dev", "watch"]
```

I could add this to the repo if helpful

**Comment by brandomr** [comment]
I will update this to handle regnets using the new export functionality in https://github.com/gyorilab/mira/pull/276

**Comment by brandomr** [comment]
This PR is updated to include the latest Mira support for regnet to JSON conversion.

**Comment by brandomr** [comment]
This PR has been rebased to `main` and now has a replacement for `DATA_SERVICE_URL` with `HMI_SERVER_URL` more or less throughout the repo.

**Pull request merged**